synthtool/protos/README.md

```diff
@@ -0,0 +1,27 @@
+# Synthtool and Autosynth Protocol
+
+**Synthtool** generates client library source code according to the currrent
+state of the source repo, and upstream repos like 
+[googleapis/googleapis](https://github.com/googleapis/googleapis).
+
+**Autosynth** periodically runs Synthtool for each client library source repo.
+Additionally, it repeatedly runs Synthtool in a binary search pattern to
+discover which upstream change triggered a change in the generate code.
+
+The proto files in this directory define the interface between.
+
+![Diagram of Synthtool and Autosynth](./diagram.png)
+
+**Synthtool** generates a file called `synth.metadata`, which contains a
+json-encoded instance of the `Metadata` class defined in [metadata.proto](./metadata.proto).  `synth.metadata` tells Autosynth which git repos were pulled
+by Synthtool while generating the library's source code.
+
+**Autosynth** consumes `synth.metadata`.  Before invoking Synthtool again, it
+observes which repos that Synthtool used during the last generation, and

```

Comment:
"observes which repos Synthtool used" or "observes repos that Synthtool used" but probably not both.

Reply:
Done.